core/src/Cuda/Kokkos_Cuda_ReduceScan.hpp

```diff
@@ -539,13 +539,18 @@ struct CudaReductionsFunctor<FunctorType, ArgTag, false, false> {
             : ((1 << width) - 1)
                   << ((threadIdx.y * blockDim.x + threadIdx.x) / width) * width;
     const int lane_id = (threadIdx.y * blockDim.x + threadIdx.x) % 32;
+
+    __syncwarp(mask);
+
     for (int delta = skip_vector ? blockDim.x : 1; delta < width; delta *= 2) {
       if (lane_id + delta < 32) {
```

Comment:
I wonder if this needs to be 
```
if ((lane_id + delta < 32) && (lane_id%(delta*2)==0)) {
```

Reply:
As I noted in chat, this suggestion didn't seem to make any different to test results, either following the other changes in this PR, or with this PR plus my volatile wrappers. In the former case, everything still passed, in the latter, the `half_t` reduction unit test still failed.

Reply:
I'm not clear on what this change in condition is supposed to achieve

Reply:
basically in the first iteration you have this:

thread read_pos write_pos
0 1 0
1 2 1
2 3 2

so threads read something which gets overwritten by the next guy in the same line.